File: backend/dotnet/ClinicAppointments.Db/Entities/Customer.cs
namespace ClinicAppointments.Db.Entities;

public class Customer
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public required string Email { get; set; }
    public required string Phone { get; set; }
    public DateOnly Birthdate { get; set; }
    public required string Cpf { get; set; }
}

File: tests/src/lib/api/validation/Customer.ts
import z from 'zod'

export const CustomerSchema = z.strictObject({
  id: z.uuidv4(),
  name: z.string(),
  email: z.email(),
  phone: z.string(),
  birthdate: z.coerce.date(),
  cpf: z.string(),
})

export type Customer = z.infer<typeof CustomerSchema>
